### PR TITLE
Improved leaves a bit more.

### LINKED ISSUE #2058: Jungle saplings drop too often.
Jungle should have a 2.5% chance drop, and not 5%.

### PR DESCRIPTION
**Pull request by bearbin**
Should fix #2058.

### CODE REVIEW

**Comment by worktycho** [comment]
Merge on CI success

**Comment by bearbin** [comment]
I think there's also something odd with shears, saplings sometimes drop when you break leaves with them, according to @NiLSPACE.

File: src/Blocks/BlockLeaves.h [hunk 1]
```diff
@@ -40,22 +40,33 @@ class cBlockLeavesHandler :
...
-					E_BLOCK_SAPLING,
-					1,
-					(m_BlockType == E_BLOCK_LEAVES) ? (a_BlockMeta & 0x03) : (2 << (a_BlockMeta & 0x01))
-				)
-			);
+			cItem(
+				E_BLOCK_SAPLING,
+				1,
+				(m_BlockType == E_BLOCK_LEAVES) ? (a_BlockMeta & 0x03) : (4 + (a_BlockMeta & 0x01))
+			));
```
Comment (madmaxoft) [inline src/Blocks/BlockLeaves.h]: The indent is wrong here. Either merge the `a_Pickups.push_back` and `cItem(` lines, or separate the two parentheses here (and indent the internals by one more level)

Comment (bearbin) [inline src/Blocks/BlockLeaves.h]: Yes, that has got messed up. I'll fix it.

**Comment by NiLSPACE** [comment]
Yes, didn't you have the shears problem as well?

**Comment by bearbin** [comment]
I think the problem is that the shears handler is calling ConvertToPickups.

https://github.com/mc-server/MCServer/blob/814e4404bb428a0d825f12e513562d4e43e9e6dc/src/Items/ItemShears.h#L29

Or maybe is the code that;s currently in ConvertToPickups supposed to be in OnBlockDestroyed for the leaves?

**Comment by bearbin** [comment]
Can anybody comment about the shears problem?

**Comment by bearbin** [comment]
Merging, the shears issue can be fixed later.

**Pull request merged**